torch/_inductor/codegen/wrapper.py

```diff
@@ -455,8 +455,14 @@ class WrapperCodeGen(CodeGen):
     Generate outer wrapper in Python that calls the kernels.
     """
 
-    def __init__(self):
+    def __init__(
+        self,
+        *,
+        custom_empty_strided_device=None,
+        device_imports: list[str] | None = None,
```

Comment:
There is an existing idiom for device imports here: https://github.com/pytorch/pytorch/blob/eb54ca7abe3820d7d0512ab13dea681a6989c780/torch/_inductor/codegen/halide.py#L1492-L1503

Can you use that ? 

You can subclass Triton Kernel and define 

```
    def codegen_kernel(self, name=None) -> str:
        code = super().codegen_kernel()
        code = insert_imports(code)
```

Reply:
Ah yes this works. The only downside is that it is repeated rather than put once but I can't imagine that would have any significant effect on the compilation.